agent/taskresource/grpcclient/credentialsfetcherclient.go

```diff
@@ -90,6 +93,71 @@ func (c CredentialsFetcherClient) AddKerberosLease(ctx context.Context, credenti
 	return credentialsFetcherResponse, nil
 }
 
+// AddNonDomainJoinedKerberosLease() invokes credentials fetcher daemon running on the host
+// to create kerberos tickets associated with gMSA accounts in domainless mode
+func (c CredentialsFetcherClient) AddNonDomainJoinedKerberosLease(ctx context.Context, credentialspecs []string, username string, password string, domain string) (CredentialsFetcherResponse, error) {
+	if len(credentialspecs) == 0 {
+		seelog.Error("credentialspecs request should not be empty")
+		return CredentialsFetcherResponse{}, nil
```

Comment:
How about following the pattern of `RenewNonDomainJoinedKerberosLease` method and return an error here? Same for the other validation below. How would the consumer know that the method failed if we don't return an error?

Reply:
sure will add it